spec/index.bs

```diff
@@ -6518,9 +6518,97 @@ and link it together with the shaders provided by the user.
 This linking is one of the reason the object is created as a whole.
 
 This combination state is created as a single object
-(by {{GPUDevice/createComputePipeline(descriptor)|GPUDevice.createComputePipeline()}} or {{GPUDevice/createRenderPipeline(descriptor)|GPUDevice.createRenderPipeline()}}),
-and switched as one
-(by {{GPUComputePassEncoder/setPipeline(pipeline)|GPUComputePassEncoder.setPipeline}} or {{GPURenderCommandsMixin/setPipeline(pipeline)|GPURenderCommandsMixin.setPipeline}} correspondingly).
+(a {{GPUComputePipeline}} or {{GPURenderPipeline}})
+and switched using one command
+({{GPUComputePassEncoder}}.{{GPUComputePassEncoder/setPipeline()}} or
+{{GPURenderCommandsMixin}}.{{GPURenderCommandsMixin/setPipeline()}} respectively).
+
+There are two ways to create pipelines:
+
+: <dfn dfn>immediate pipeline creation</dfn>
+:: {{GPUDevice/createComputePipeline()}} and {{GPUDevice/createRenderPipeline()}}
+    return a pipeline object which can be used immediately in a pass encoder.
+
+    When this fails, the pipeline object will be invalid and the call will generate either a
+    [$validation error$] or an [$internal error$].
+
+    Note: If pipeline creation takes a long time, this can incur a stall in the
+    [=device timeline=] at some point between the creation call and execution of the
+    {{GPUQueue/submit()}} in which it is first used.
+    The point is unspecified, but most likely to be one of: at creation, at the first usage of the
+    pipeline in `setPipeline()`, at the corresponding `finish()` of that {{GPUCommandEncoder}} or
+    {{GPURenderBundleEncoder}}, or at {{GPUQueue/submit()}} of that {{GPUCommandBuffer}}.
+
+: <dfn dfn>promised pipeline creation</dfn>
+:: {{GPUDevice/createComputePipelineAsync()}} and {{GPUDevice/createRenderPipelineAsync()}}
+    return a `Promise` which resolves to a pipeline object when creation of the pipeline has
+    completed.
+
+    When this fails, the `Promise` rejects with a {{GPUPipelineError}}.
+
+<dfn interface>GPUPipelineError</dfn> describes a pipeline creation failure.
+
+<script type=idl>
+[Exposed=(Window, DedicatedWorker), SecureContext, Serializable]
+interface GPUPipelineError : DOMException {
+    constructor(DOMString message, GPUPipelineErrorInit options);
+    readonly attribute GPUPipelineErrorReason reason;
```

Comment:
Just repeating here what I already said on the call: This seems like a perfectly reasonable approach for the time being, and I'm satisfied that it doesn't prevent exposing more detailed error information in the future in the unlikely event that we decide that there's something useful here that we can consistently expose across all backends. 👍